### PR TITLE
Merge upstream

### PR DESCRIPTION
**Pull request by spd-intermodalics**
## Description

This PR is a merge of upstream https://github.com/osrf/rocker `main`.

It contains upstream patches until https://github.com/osrf/rocker/commit/2b8d5abb18c829d22f290679b739dc53765198ad (15th May 2023).

## Conflict resolution

Since upstream uses tags for the os detection, and that is main different with this repository, upstream introduced image cleaning after the check. In this merge, that is removed, so the image detector image will remain until a pruning is made.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +1,12 @@
 [sdist_dsc]
 epoch: 1
-debian-version: 2intermodalics
+debian-version: 3intermodalics
```
Comment (meyerj) [inline setup.cfg]: nit: As the main version was updated from `0.2.6` to `0.2.12` in the meantime, the debian version should be reset to `1intermodalics` (according to the [Debian Policy Manual](https://www.debian.org/doc/debian-policy/ch-controlfields.html#version)). The debian version (or revision) is usually only increased when a new release is made for the same upstream version.

**Review by meyerj — APPROVED** [review]
Did not test it myself, but if you did and also tested the `--user` extension (we still had some changes compared to upstream there), it should be fine.

**Comment by spd-intermodalics** [comment]
The option `--user` still works for a couple of examples, so I think it should be fine.

**Comment by meyerj** [comment]
Pushed the new debian to http://packages.intermodalics.eu/intermodalics.

**Pull request merged**